### ISSUE TITLE
Default exception handler does not log exceptions (but prints to stdout)

### ISSUE THREAD

**Issue by briankontor** [metosin/compojure-api#172]
We noticed that uncaught exceptions in our api were not being logged and found that the default exception handler `safe-handler` calls `printStackTrace` instead of logging the exception.

A small but useful change might be something like:
```
(defn safe-handler
  "Copy of the version in compojure api that uses log instead of printStackTrace
  See https://github.com/metosin/compojure-api/blob/master/src/compojure/api/exception.clj#L13"
  [^Exception e _ _]
  (clojure.tools.logging/spy :error e) ;; was (.printStackTrace e)
  (internal-server-error {:type "unknown-exception"
                          :class (.getName (.getClass e))}))
```

**Comment by ikitommi** [comment]
Good point. We started with "plug in your own implementations for everything", but the defaults should be better. Will fix this.

**Comment by ikitommi** [comment]
don't want to add another dependency, but we could route all internal logging via something like this: https://github.com/metosin/compojure-api/compare/clojure.tools.logging?expand=1

what do you think?

**Comment by ikitommi** [comment]
This should be fixed in `0.24.1`.

**Comment by briankontor** [comment]
@ikitommi Looks good, thanks!
